File: src/main/java/com/intellij/aws/cloudformation/CloudFormationConstants.java
package com.intellij.aws.cloudformation;

import com.intellij.util.containers.HashSet;

import java.util.Arrays;
import java.util.Set;

public class CloudFormationConstants {
  public static final Set<String> SupportedTemplateFormatVersions = new HashSet<String>(Arrays.asList(
    "2010-09-09"
  ));

  public static final String CustomResourceType = "AWS::CloudFormation::CustomResource";
  public static final String CustomResourceTypePrefix = "Custom::";

  public static final String CommentResourcePropertyName = "Comment";

  public static final String ConditionPropertyName = "Condition";
  public static final String TypePropertyName = "Type";
  public static final String PropertiesPropertyName = "Properties";
  public static final String CreationPolicyPropertyName = "CreationPolicy";
  public static final String DeletionPolicyPropertyName = "DeletionPolicy";
  public static final String DependsOnPropertyName = "DependsOn";
  public static final String MetadataPropertyName = "Metadata";
  public static final String UpdatePolicyPropertyName = "UpdatePolicy";
  public static final String VersionPropertyName = "Version";

  public static final Set<String> AllTopLevelResourceProperties = new HashSet<String>(Arrays.asList(
    ConditionPropertyName,
    TypePropertyName,
    PropertiesPropertyName,
    CreationPolicyPropertyName,
    DeletionPolicyPropertyName,
    DependsOnPropertyName,
    MetadataPropertyName,
    UpdatePolicyPropertyName,
    VersionPropertyName
  ));
}

File: src/main/kotlin/com/intellij/aws/cloudformation/CloudFormationConstants.kt
package com.intellij.aws.cloudformation

import com.intellij.util.containers.HashSet
import java.util.Arrays

object CloudFormationConstants {
  val SupportedTemplateFormatVersions: Set<String> = HashSet(Arrays.asList("2010-09-09"))

  val CloudFormationInterfaceType = "AWS::CloudFormation::Interface"
  val CloudFormationInterfaceParameterLabels = "ParameterLabels"
  val CloudFormationInterfaceParameterGroups = "ParameterGroups"
  val CloudFormationInterfaceParameters = "Parameters"

  val CustomResourceType = "AWS::CloudFormation::CustomResource"
  val CustomResourceTypePrefix = "Custom::"

  val CommentResourcePropertyName = "Comment"

  val ConditionPropertyName = "Condition"
  val TypePropertyName = "Type"
  val PropertiesPropertyName = "Properties"
  val CreationPolicyPropertyName = "CreationPolicy"
  val DeletionPolicyPropertyName = "DeletionPolicy"
  val DependsOnPropertyName = "DependsOn"
  val MetadataPropertyName = "Metadata"
  val UpdatePolicyPropertyName = "UpdatePolicy"
  val VersionPropertyName = "Version"

  val AllTopLevelResourceProperties = listOf(
      ConditionPropertyName,
      TypePropertyName,
      PropertiesPropertyName,
      CreationPolicyPropertyName,
      DeletionPolicyPropertyName,
      DependsOnPropertyName,
      MetadataPropertyName,
      UpdatePolicyPropertyName,
      VersionPropertyName
  ).toSet()
}
